File: client/src/ts/packets/movement.ts
import Packet, {PacketType} from "./packet";
import Vector2D from "../vector2d";

export default class MovementPacket extends Packet {
    constructor(private newLocation: Vector2D) {
        super(PacketType.Movement);
    }

    public dictify(): any {
        return {'type': this.packetType, 'x': this.newLocation.x, 'y': this.newLocation.y};
    }
}

File: server/network/packets/movement.py
from server.network.packets import PacketType
from server.network.packets.packet import Packet


class MovementPacket(Packet):
    def __init__(self, data):
        if 'direction' not in data:
            print(data)
        self.direction = data['direction']

    def dictify(self):
        pass

    def get_type(self):
        return PacketType.Movement
